From 69d12201a0ee50bb5646ddc734574b6664587905 Mon Sep 17 00:00:00 2001
From: Chiqijun <chiqijun@huawei.com>
Date: Tue, 2 Mar 2021 20:07:44 +0800
Subject: [PATCH] net/hinic: Remove the initialization of the global variable
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 69d12201a0ee50bb5646ddc734574b6664587905

 g_uld_info

driver inclusion
category: bugfix
bugzilla: 47993

-----------------------------------------------------------------------

In the hinic driver, the g_uld_info global variable is assigned to all
0s, and the'designated init' alarm is reported at compile time. The
global variable defaults to all 0s, and the assignment of 0s is not
necessary.

Signed-off-by: Chiqijun <chiqijun@huawei.com>
Reviewed-by: Wangxiaoyun <cloud.wangxiaoyun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/huawei/hinic/hinic_lld.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/huawei/hinic/hinic_lld.c b/drivers/net/ethernet/huawei/hinic/hinic_lld.c
index fddb949ff9aa..6c960cecf101 100644
--- a/drivers/net/ethernet/huawei/hinic/hinic_lld.c
+++ b/drivers/net/ethernet/huawei/hinic/hinic_lld.c
@@ -130,7 +130,7 @@ struct hinic_pcidev {
 #define MAX_CARD_ID 64
 static u64 card_bit_map;
 LIST_HEAD(g_hinic_chip_list);
-struct hinic_uld_info g_uld_info[SERVICE_T_MAX] = { {0} };
+struct hinic_uld_info g_uld_info[SERVICE_T_MAX];
 static const char *s_uld_name[SERVICE_T_MAX] = {
 	"nic", "ovs", "roce", "toe", "iwarp", "fc", "fcoe", "migrate"};
 
-- 
2.26.2

